packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -501,28 +502,14 @@ export default {
           } catch (error) {
             if (/Unsupported node type/.test(error.message)) {
               if (declaredDependencyNode.type === 'Literal') {
-                if (typeof declaredDependencyNode.value === 'string') {
-                  context.report({
-                    node: declaredDependencyNode,
-                    message:
-                      `The ${
-                        declaredDependencyNode.raw
-                      } string literal is not a valid dependency ` +
-                      `because it never changes. Did you mean to ` +
-                      `include ${
-                        declaredDependencyNode.value
-                      } in the array instead?`,
-                  });
-                } else {
-                  context.report({
-                    node: declaredDependencyNode,
-                    message:
-                      `The '${
-                        declaredDependencyNode.raw
-                      }' literal is not a valid dependency ` +
-                      'because it never changes. You can safely remove it.',
-                  });
-                }
+                context.report({
```

Comment:
I'm not 100% sure about removing the string special case. I understand people could be confused by the normal message? But I'm worried that it's confusing in the case that it doesn't correspond to an actual variable. (Maybe we could check that it looks like a valid identifier/member expression?)

Reply:
I think ideally we'd special case values that are in scope in this message.